beater/beater.go

```diff
@@ -255,13 +259,20 @@ func (s *serverCreator) Create(p beat.PipelineConnector, rawConfig *common.Confi
 	}
 	apmServerCommonConfig := integrationConfig.APMServer
 	apmServerCommonConfig.Merge(common.MustNewConfigFrom(`{"data_streams.enabled": true}`))
-	return newServerRunner(s.context, serverRunnerParams{
+	params := serverRunnerParams{
 		sharedServerRunnerParams: s.args,
 		Namespace:                namespace,
 		Pipeline:                 p,
 		RawConfig:                apmServerCommonConfig,
 		FleetConfig:              &integrationConfig.Fleet,
-	})
+	}
+	if s.runner != nil {
+		// If we've already created a runner, just reconfigure it.
+		return s.runner, s.runner.configure(params)
+	} else {
+		s.runner, err = newServerRunner(s.context, params)
+		return s.runner, err
+	}
```

Comment:
The serverCreator holds a reference to the runner; when it receives a new config it parses that (as it does now), but instead of creating a new runner it will pass the params so it can update it's configuration.